## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 430088`

Logs: https://github.com/liberodark/nixpkgs-review-gha/actions/runs/16674581499


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ucon64</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>ucon64</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>ucon64</summary>
<pre>      |           ~~~~~~~~~~~~~~~^~~~
misc/parallel.c:449:46: warning: unused parameter 'byte' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-parameter-Wunused-parameter8;;]
  449 | outportb (unsigned short port, unsigned char byte)
      |                                ~~~~~~~~~~~~~~^~~~
misc/parallel.c: In function 'outportw':
misc/parallel.c:521:26: warning: unused parameter 'port' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-parameter-Wunused-parameter8;;]
  521 | outportw (unsigned short port, unsigned short word)
      |           ~~~~~~~~~~~~~~~^~~~
misc/parallel.c:521:47: warning: unused parameter 'word' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-unused-parameter-Wunused-parameter8;;]
  521 | outportw (unsigned short port, unsigned short word)
      |                                ~~~~~~~~~~~~~~~^~~~
misc/parallel.c: In function 'inportb':
misc/parallel.c:381:1: warning: control reaches end of non-void function [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-return-type-Wreturn-type8;;]
  381 | }
      | ^
misc/parallel.c: In function 'inportw':
misc/parallel.c:445:1: warning: control reaches end of non-void function [8;;https://gcc.gnu.org/onlinedocs/gcc-14.3.0/gcc/Warning-Options.html#index-Wno-return-type-Wreturn-type8;;]
  445 | }
      | ^
make: *** [Makefile:174: misc/parallel.o] Error 1</pre>
</details>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ucon64</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ucon64</li>
  </ul>
</details>

--- a/pkgs/by-name/uc/ucon64/package.nix
+++ b/pkgs/by-name/uc/ucon64/package.nix
@@ -1,0 +1,58 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  versionCheckHook,
+  libusb1,
+  zlib,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ucon64";
+  version = "2.2.2";
+
+  src = fetchurl {
+    url = "https://sourceforge.net/projects/ucon64/files/ucon64/ucon64-${finalAttrs.version}/ucon64-${finalAttrs.version}-src.tar.gz/download";
+    name = "ucon64-${finalAttrs.version}-src.tar.gz";
+    hash = "sha256-4QCtSjD2wZq96Y42HGoOysTkBHf1TPt1SYxczSH7Ohg=";
+  };
+
+  buildInputs = [
+    zlib
+    libusb1
+  ];
+
+  sourceRoot = "ucon64-${finalAttrs.version}-src/src";
+
+  # Disable parallel on ARM (sys/io.h is x86-only)
+  configureFlags = lib.optionals (!stdenv.hostPlatform.isx86) [
+    "--disable-parallel"
+  ];
+
+  makeFlags = [
+    "CC=${stdenv.cc.targetPrefix}cc"
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    install -Dm755 ucon64 -t $out/bin/
+
+    runHook postInstall
+  '';
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+
+  versionCheckProgramArg = "--version";
+  doInstallCheck = true;
+
+  meta = {
+    description = "Tool to backup video game cartridges and perform various ROM operations";
+    homepage = "https://ucon64.sourceforge.io/";
+    license = lib.licenses.gpl2Plus;
+    maintainers = with lib.maintainers; [ liberodark ];
+    mainProgram = "ucon64";
+  };
+})

